docs/elasticsearch/search-template.md

```diff
@@ -0,0 +1,416 @@
+---
+layout: default
+title: Search Templates
+parent: Elasticsearch
+nav_order: 11
+---
+
+# Search templates
+
+You can convert your full-text queries into a search template to accept user input and dynamically insert it into your query.
+
+For example, if you use Elasticsearch as a backend search engine for your application or website, you can take in user queries from a search bar or a form field and pass them as parameters into a search template. That way, the syntax to create Elasticsearch queries is abstracted from your end users.
+
+Whenever you find yourself writing code to convert user input into Elasticsearch queries, you can simplify your code with search templates. If you need to add fields to your search query, you can just modify the template without making changes to your code.
+
+Search templates are expressed using the Mustache language. For a list of all syntax options, see the [Mustache manual](http://mustache.github.io/mustache.5.html).
+{: .note }
+
+---
+
+#### Table of contents
+1. TOC
+{:toc}
+
+## Create search templates
+
+A search template has two components: the query and the parameters. Parameters are user inputted values that get placed into variables. Variables are represented with double braces in Mustache notation. When encountering a variable like `{% raw %}{{var}}{% endraw %}` in the query, Elasticsearch goes to the `params` section and looks for a parameter called `var` and replaces it with the specified value.
```

Comment:
Elasticsearch goes to the `params` section, looks for a parameter called `var`, and replaces it with the specified value.